### PR TITLE
Add Prometheus exporter to ROCK

### PR DESCRIPTION
**Pull request by gtrkiller**
Applicable spec: [ISD043](https://docs.google.com/document/d/1H9jC6hSAo5eKli5FWLDUKpiP1MjzIfCFeA74vfI2JrM)

### Overview

Add the NGINX Prometheus Exporter as a layer to the content-cache ROCK.

### Rationale

Simplify the charm, reduce container amount, address possible securities issues caused by docker images.

### Juju Events Changes

N/A

### Module Changes

N/A

### Library Changes

N/A

### Checklist

- [X] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [X] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [X] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [X] The documentation is generated using `src-docs`
- [X] The documentation for charmhub is updated
- [X] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

**Review by amandahla — DISMISSED** [review]
LGTM

File: content-cache_rock/rockcraft.yaml [hunk 1]
```diff
@@ -15,6 +15,8 @@ parts:
       - nginx-light
       - bash
       - coreutils
+    stage-snaps:
+      - gtrkiller-nginx-prometheus-exporter/latest/edge
```
Comment (jdkandersson) [inline content-cache_rock/rockcraft.yaml:19]: Should this include your handle? Or should it be more general?

Comment (gtrkiller) [reply content-cache_rock/rockcraft.yaml:19]: given that there is already an NGINX Prometheus Exporter without the handle, but that we cannot edit at  all https://snapcraft.io/nginx-prometheus-exporter I think leaving my handle there for now is useful to not confuse my snap with the other one.

**Review by arturo-seijas — APPROVED** [review]

**Review by amandahla — APPROVED** [review]
LGTM

**Pull request merged**